pkg/utils/path.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package utils
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strings"
+)
+
+type FPath struct {
+	local  bool
+	schema string
+	path   string
+}
+
+// Creates new Struct from handed over URL
+func (p *FPath) New(url string) {
+
+	// Check for Schema
+	sepstring := strings.Split(url, "://")
+
+	switch len(sepstring) {
+	case 2: // Has Schema
+		p.schema = sepstring[0]
+		p.path = sepstring[1]
+	case 1: // No Schema
+		p.local = true
+		p.path = sepstring[0]
+	default: // Everything else is misformatted
+		log.Fatalf("misformatted URL: %v", url)
+	}
+
+	// Check for Windows Special Handling Prefix
+	cprefix, _ := regexp.Compile(`^\\\\\?\\(UNC\\)?`)
+
+	// when Prefix present, omit further changes to the path
+	if prefix := cprefix.FindString(p.path); prefix != "" {
+		p.schema = prefix
+		p.path = strings.Replace(p.path, prefix, "", -1) //Strip Prefix
+	} else {
+		// when file is local, ensure path absolute
+		if p.IsLocal() && !filepath.IsAbs(p.path) {
+			p.path, _ = filepath.Abs(p.path)
```

Comment:
filepath.Abs may fail... check the error and only update the p.path when there is no error.